sydent/sydent.py

```diff
@@ -291,7 +297,7 @@ def parse_config_dict(config_dict):
         config_dict (dict): the configuration dictionary to be parsed
     """
     # Build a config dictionary from the defaults merged with the given dictionary
-    config = CONFIG_DEFAULTS
+    config = copy.deepcopy(CONFIG_DEFAULTS)
```

Comment:
why?

Reply:
Because otherwise we're editing the defaults directly, which is mostly an issue with tests, because then if you run a test with a specific value for a setting and then another one relying on that setting's default value, then that second test will fail because the default value has been changed by the first test.

Reply:
This was actually causing the tests to fail, because `ThreepidInvitesNoDeleteTestCase` would run before `ThreepidInvitesTestCase` and would change the default value for `delete_tokens_on_bind` to `false` instead of `true`, thus making the latter fail.

Reply:
oh! yes. sorry. duh.